interface.py

```diff
@@ -15,110 +15,113 @@ def __init__(self, dealt_cards, robot_data):
         self.my_program = [None, None, None, None, None]
         self.power_down = False
         self.indicator = False
-        self.cursor_index = 0 # 0-4 number of positon
+        self.cursor_index = 0  # 0-4 number of positon
 
     def __repr__(self):
-        return "<InterfaceState Cards: {}, My Cards: {}, Power Down: {}, Robot: {}>".format(self.dealt_cards, self.my_program, self.power_down, self.robot_data)
+        return f"InterfaceState Cards: {self.dealt_cards}, My Cards: {self.my_program}, Power Down: {self.power_down,}, Robot: {self.robot_data}"
 
     def select_card(self, dealt_card_index):
         """
         Select a card from the dealt cards and put on
         a place where selector is (in the robot program)
         and selector cursor moves to the next free place.
         """
-        if self.indicator == False:
+        if self.indicator is False:
```

Comment:
jaký je mezi tím prosím rozdíl, jestli `==` nebo `is`?

Reply:
Když ses na to zeptala, napadlo mě, že vlastně celé to tvrzení je špatně postavené - mělo by spíš být `if not self.indicator` (opravím, až budu mít chvíli). Ale jak to cítím, je to o rozdílu mezi totožnosti a porovnání (jestli je to tentýž objekt, nebo "jen" objekt o stejné hodnotě) - všechny True jsou vždy tentýž objekt, nevytváří se nové "instance pravdy" a "instance nepravdy" pro False. Proto je logičtější ptát se u True/False/None na `is` místo `==`. A podle toho to kontroluje Flake8, kontrola stylu Pythonu. https://www.geeksforgeeks.org/difference-operator-python/